### PR TITLE
Allow remote query items to have their labels edited

### PR DESCRIPTION
**Pull request by aeisenberg**
The labels for remote query items are interpolated using the same
strategy as local queries with two caveats:

1. There is no easy way to get the result count without reading files,
   so, this value is kept empty.
2. There is no database name for remote queries. Instead, use the 
   nwo of the controller repo.

### CODE REVIEW

**Comment by charisk** [comment]
I've unfortunately not been able to look at this properly today. From a quick glance, it looks like there was some refactoring done as well as new functionality being added. If that's the case, would it be possible to split into two PRs, one that does the refactoring and one that introduces the new functionality?

**Comment by aeisenberg** [comment]
I split the PR into two commits. The first for the refactoring and the second for new behaviour and tests. I hope this is easier for you to understand.

File: extensions/ql-vscode/src/history-item-label-provider.ts [hunk 1]
```diff
@@ -60,4 +62,20 @@ export class HistoryItemLabelProvider {
...
+  private getRemoteInterpolateReplacements(item: RemoteQueryHistoryItem): InterpolateReplacements {
+    return {
+      t: new Date(item.remoteQuery.executionStartTime).toLocaleString(env.language),
+      q: item.remoteQuery.queryName,
+
+      // There is no database name for remote queries. Instead use the controller repository name.
+      d: `${item.remoteQuery.controllerRepository.owner}/${item.remoteQuery.controllerRepository.name}`,
+
+      // There is no synchronous way to get the results count.
+      r: '',
```
Comment (shati-patel) [inline extensions/ql-vscode/src/history-item-label-provider.ts:75]: I think it's fine to have the result count empty, but perhaps we should also change the default string? It looks a bit strange to have `   result count` in the default label, even for remote queries:

image

**Review by shati-patel — COMMENTED** [review]
Thanks for doing this! 💖 I ended up looking at the [first 2 commits](https://github.com/github/vscode-codeql/pull/1290/files/eec506a209b3b74233dd7743d482edd290d1ce3d) for the refactoring + tests. And then the [last commit](https://github.com/github/vscode-codeql/pull/1290/commits/f1c4fef8ba88b453a668455083e9c3f4eb59fcf6) for new remote queries labelling, so I think I've understood the changes now 😄 

Just a few small comments—mainly LGTM 👍🏽 

(reviewed with @charisk)

File: extensions/ql-vscode/src/history-item-label-provider.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
+import { env } from 'vscode';
+import { QueryHistoryConfig } from './config';
+import { LocalQueryInfo, QueryHistoryInfo } from './query-results';
+import { RemoteQueryHistoryItem } from './remote-queries/remote-query-history-item';
+
+interface InterpolateReplacements {
+  t: string; // Start time
+  q: string; // Query name
+  d: string; // Database/List name
```
[the remark this answers is not in the document]
Comment (aeisenberg) [reply extensions/ql-vscode/src/history-item-label-provider.ts:9]: For now, let's just use the controller repo, even though list name would be more useful. In the future, we can make the list name more easily accessible. I will create an issue for that for future work.

File: extensions/ql-vscode/src/history-item-label-provider.ts (hunk 1 above)
Comment (aeisenberg) [reply extensions/ql-vscode/src/history-item-label-provider.ts:75]: Yes. That does look odd. Not sure what to put there...

File: extensions/ql-vscode/src/query-history.ts [hunk 3]
```diff
@@ -647,19 +653,21 @@ export class QueryHistoryManager extends DisposableObject {
...
+    if (!this.assertSingleQuery(finalMultiSelect)) {
       return;
     }
 
     const response = await window.showInputBox({
-      prompt: 'Label:',
-      placeHolder: '(use default)',
-      value: finalSingleItem.label,
+      // prompt: 'Label:',
+      placeHolder: `(use default: ${this.queryHistoryConfigListener.format})`,
```
[the remark this answers is not in the document]
Comment (aeisenberg) [reply extensions/ql-vscode/src/query-history.ts:662]: Yep....I'll remove.

**Review by shati-patel — APPROVED** [review]
Thanks for the changes! :shipit:

**Pull request merged**